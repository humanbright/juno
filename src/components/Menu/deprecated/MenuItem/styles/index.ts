export * from './StyledMenuItem';
export * from './StyledMuiListItemIcon';
export * from './StyledRcListItemLabel';
