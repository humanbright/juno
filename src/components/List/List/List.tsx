import MuiList from '@material-ui/core/List';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../../foundation';
import { ListStyle } from './styles';
import { RcListClasses } from './utils';

type RcListProps = {} & RcBaseProps<ComponentProps<typeof MuiList>>;

const _RcList = forwardRef<any, RcListProps>((inProps, ref) => {
  const props = useThemeProps({ props: inProps, name: 'RcList' });
  const { classes: classesProp, children, ...rest } = props;
  const classes = useMemo(
    () => combineClasses(RcListClasses, classesProp),
    [classesProp],
  );

  return (
    <MuiList {...rest} ref={ref} classes={classes}>
      {children}
    </MuiList>
  );
});

const RcList = styled(_RcList)`
  ${ListStyle}
`;

RcList.defaultProps = {
  disablePadding: true,
};

RcList.displayName = 'RcList';

export { RcList };
export type { RcListProps };
