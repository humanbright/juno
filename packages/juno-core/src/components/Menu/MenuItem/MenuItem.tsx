import React, {
  ComponentProps,
  ElementType,
  forwardRef,
  ReactElement,
  ReactNode,
  Ref,
  useContext,
  useImperativeHandle,
  useMemo,
} from 'react';

import clsx from 'clsx';

import MuiMenuItem from '@material-ui/core/MenuItem';

import {
  Classes,
  combineClasses,
  isRcElement,
  logInDev,
  omit,
  RcBaseProps,
  RcBaseSize,
  RcClassesProps,
  styled,
  useEventCallback,
  useId,
  useThemeProps,
} from '../../../foundation';
import { Check as checkSvg } from '@ringcentral/juno-icon';
import { RcIcon, RcIconProps } from '../../Icon';
import { RcListItemSecondaryAction } from '../../List';
import { RcListItemAvatar } from '../../List/ListItemAvatar';
import { RcListItemIcon } from '../../List/ListItemIcon';
import { withTooltip, WithTooltipProps } from '../../Tooltip';
import { RcMenuContext } from '../Menu/MenuContext';
import { RcMenuListContext } from '../MenuList/MenuListContext';
import { RcSubMenuContext } from '../SubMenu/SubMenuContext';
import { MenuItemStyle, StyledCheckIcon } from './styles';
import { RcMenuItemClasses } from './utils';

type RcMenuItemSize = RcBaseSize<'large' | 'medium'>;

type RcMenuItemType = 'checked' | 'selected';

type RcMenuItemClassesType = RcClassesProps<
  Classes<ComponentProps<typeof MuiMenuItem>> | 'checked' | 'unchecked'
>;

type RcMenuItemInnerProps = {
  /** id ref, can get menu item id */
  idRef?: Ref<string>;
  /** for subMenu component, use prevent click */
  isSubMenuItem?: boolean;
};

type RcMenuItemProps = {
  /** MenuItem size */
  size?: RcMenuItemSize;
  /** The component used for the root node. Either a string to use a HTML element or a component. */
  component?: ElementType;
  /** checked for that should render checked icon */
  type?: RcMenuItemType;
  /** if type to be `checked`, the checked prop for that should render checked icon */
  checked?: boolean;
  /** MenuItem with icon, can use `RcListItemIcon` */
  icon?: ReactNode;
  /** pass to the icon, can use RcIcon with symbol prop */
  symbol?: RcIconProps['symbol'];
  /** MenuItem with avatar, can use ListItemAvatar */
  avatar?: ReactElement;
  /** MenuItem with subAction, can use ListItemSecondaryAction */
  secondaryAction?: ReactNode;
} & RcMenuItemClassesType &
  WithTooltipProps &
  RcBaseProps<ComponentProps<typeof MuiMenuItem>, 'classes' | 'title'>;

const _RcMenuItem = forwardRef<any, RcMenuItemProps & RcMenuItemInnerProps>(
  (inProps, ref) => {
    const props = useThemeProps({ props: inProps, name: 'RcMenuItem' });
    const {
      classes: classesProp,
      children,
      className,
      onMouseEnter,
      onClick,
      onFocus,
      size,
      type,
      checked,
      icon,
      symbol,
      avatar,
      secondaryAction,
      idRef,
      isSubMenuItem,
      title,
      ...rest
    } = props;

    const isCheckedType = type === 'checked';
    const menuItemId = useId('menu-item', true);

    const menuContext = useContext(RcMenuContext);
    const menuListContext = useContext(RcMenuListContext);
    const subMenuContext = useContext(RcSubMenuContext);

    const setFocusedMenuItemId = useEventCallback(() => {
      if (!menuItemId) return;

      if (menuListContext.menuListId) {
        menuListContext.setFocusedMenuItemId(menuItemId);
      } else if (menuContext.menuId) {
        menuContext.setFocusedMenuItemId(menuItemId);
      }
    });

    const classes = useMemo(
      () => combineClasses(RcMenuItemClasses, classesProp),
      [classesProp],
    );

    const toClasses = useMemo(
      () => omit(classes, ['checked', 'unchecked']),
      [classes],
    );

    const itemAvatar = useMemo(() => {
      if (React.isValidElement(avatar)) {
        return isRcElement(avatar, ['RcListItemAvatar']) ? (
          avatar
        ) : (
          <RcListItemAvatar>{avatar}</RcListItemAvatar>
        );
      }
      return null;
    }, [avatar]);

    const itemIcon = useMemo(() => {
      if (symbol) {
        return (
          <RcListItemIcon>
            <RcIcon size="small" symbol={symbol} />
          </RcListItemIcon>
        );
      }

      if (React.isValidElement(icon)) {
        return isRcElement(icon, ['RcListItemIcon']) ? (
          icon
        ) : (
          <RcListItemIcon>{icon}</RcListItemIcon>
        );
      }

      if (typeof icon === 'string') {
        logInDev({
          component: 'RcMenuItem',
          message: 'please use symbol to replace icon string',
        });
        return (
          <RcListItemIcon>
            <RcIcon size="small">{icon}</RcIcon>
          </RcListItemIcon>
        );
      }

      return null;
    }, [icon, symbol]);

    const itemSubAction = useMemo(() => {
      if (isCheckedType && checked) {
        return (
          <RcListItemSecondaryAction>
            <StyledCheckIcon
              size="large"
              symbol={checkSvg}
              color={['interactive', 'f01']}
            />
          </RcListItemSecondaryAction>
        );
      }

      return secondaryAction ? (
        isRcElement(secondaryAction, ['RcListItemSecondaryAction']) ? (
          secondaryAction
        ) : (
          <RcListItemSecondaryAction>
            {secondaryAction}
          </RcListItemSecondaryAction>
        )
      ) : null;
    }, [checked, isCheckedType, secondaryAction]);

    const handleMouseEnter = useEventCallback(
      (e: React.MouseEvent<HTMLLIElement>) => {
        onMouseEnter?.(e);
        setFocusedMenuItemId();
      },
    );

    const handleClick = useEventCallback(
      (e: React.MouseEvent<HTMLLIElement>) => {
        onClick?.(e);

        if (menuContext.autoClose && !isSubMenuItem) {
          menuContext.closeMenu(e, 'itemClick');
        }
        if (subMenuContext.autoClose && !isSubMenuItem) {
          subMenuContext?.closeSubMenu?.(e, 'itemClick');
        }
      },
    );

    const handleFocus = useEventCallback(
      (e: React.FocusEvent<HTMLLIElement>) => {
        setFocusedMenuItemId();
        onFocus?.(e);
      },
    );

    useImperativeHandle(idRef, () => menuItemId, [menuItemId]);

    return (
      <MuiMenuItem
        aria-checked={isCheckedType && checked ? true : undefined}
        {...rest}
        ref={ref as any}
        title={typeof title === 'string' ? title : undefined}
        classes={toClasses}
        className={clsx(className, {
          [classes!.checked]: isCheckedType && checked,
          [classes!.unchecked]: isCheckedType && !checked,
        })}
        onMouseEnter={handleMouseEnter}
        onClick={handleClick}
        onFocus={handleFocus}
        data-menuitem-id={menuItemId}
      >
        {itemAvatar}
        {itemIcon}
        {children}
        {itemSubAction}
      </MuiMenuItem>
    );
  },
);

const RcMenuItem = styled(withTooltip(_RcMenuItem))`
  ${MenuItemStyle};
`;

RcMenuItem.defaultProps = {
  size: 'medium',
};

RcMenuItem.displayName = 'RcMenuItem';

export { RcMenuItem };
export type {
  RcMenuItemClassesType,
  RcMenuItemProps,
  RcMenuItemSize,
  RcMenuItemType,
};
