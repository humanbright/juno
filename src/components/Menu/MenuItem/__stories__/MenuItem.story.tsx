import React, { ComponentProps } from 'react';

import { Meta, Story } from '@storybook/react';

import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
  switchToControlKnobs,
} from '../../../../storybook';
import { RcListItemText } from '../../../List';
import { RcMenuItem } from '../MenuItem';

export default {
  title: 'ðŸš€ Cleanup Components/Menu/MenuItem',
  component: RcMenuItem,
  argTypes: {
    ...sortInDocTable<keyof MenuItemProps>([
      'component',
      'size',
      'selected',
      'type',
      'checked',
      'disabled',
      'divider',
      'dense',
      'alignItems',
      'disableGutters',
      'tabIndex',
      'autoFocus',
      'icon',
      'symbol',
      'avatar',
      'secondaryAction',
    ]),
    ...notControlInDocTable<keyof MenuItemProps>([]),
    ...notShowInDocTable<keyof MenuItemProps>(['action', 'isSubMenuItem']),
  },
} as Meta;

type MenuItemProps = ComponentProps<typeof RcMenuItem>;

export const MenuItem: Story<MenuItemProps> = ({ children, ...args }) => {
  switchToControlKnobs();
  return (
    <>
      <RcMenuItem {...args}>
        <RcListItemText primary="MenuItem content" />
      </RcMenuItem>
      <br />
      <RcMenuItem {...args}>
        <RcListItemText
          primary="MenuItem content"
          secondary="Secondary content"
        />
      </RcMenuItem>
    </>
  );
};

MenuItem.args = {};

MenuItem.argTypes = {
  ...notControlInDocTable<keyof MenuItemProps>([]),
};

MenuItem.parameters = {
  tags: [
    {
      name: 'Mui',
      href: 'https://material-ui.com/components/menus/',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};

MenuItem.storyName = 'MenuItem';
