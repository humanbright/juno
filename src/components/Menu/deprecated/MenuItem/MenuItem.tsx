import { MenuItemProps as MuiMenuItemProps } from '@material-ui/core/MenuItem';
import clsx from 'clsx';
import uniqueId from 'lodash/uniqueId';
import React, {
  forwardRef,
  ReactNode,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';

import {
  Classes,
  combineClasses,
  omit,
  RcBaseProps,
  RcBaseSize,
  RcClassesProps,
  styled,
  useEventCallback,
  withDeprecatedCheck,
} from '../../../../foundation';
import checkSvg from '../../../../icon/Check';
import { RcIcon, RcIconProps } from '../../../Icon';
import { withTooltip } from '../../../Tooltip';
import { RcMenuContext } from '../Menu';
import { RcMenuListContext } from '../MenuList';
import { RcSubMenuContext } from '../SubMenu';
import {
  StyledMenuItem,
  StyledMuiListItemIcon,
  StyledRcIcon,
  StyledRcListItemLabel,
} from './styles';
import { RcMenuItemClasses } from './utils/MenuItemUtil';

type MenuItemSize = RcBaseSize<'large' | 'medium'>;

type RcMenuItemType = 'checked' | 'selected';

type RcMenuItemClassesType = RcClassesProps<
  Classes<MuiMenuItemProps> | 'checked' | 'unchecked'
>;

type RcMenuItemProps = {
  /** MenuItem size */
  size?: MenuItemSize;
  /** MenuItem with subtext */
  subtext?: ReactNode;

  /** checked for that should render checked icon */
  type?: RcMenuItemType;
  /** if type to be `checked`, the checked prop for that should render checked icon */
  checked?: boolean;

  /** @deprecated  MenuItem with icon, pls use RcListItemIcon */
  icon?: ReactNode;
  /** @deprecated pass to the icon, pls use RcIcon with symbol prop */
  symbol?: RcIconProps['symbol'];
  /** @deprecated MenuItem with avatar, pls use ListItemAvatar */
  avatar?: React.ReactElement;
  /** @deprecated */
  label?: string;

  /** @deprecated the maxWidth of MenuItem, recommend pass classes to define it. */
  maxWidth?: number;
  /**
   * @deprecated
   * when hover on MenuItem, is cursor in shape of pointer.
   * Instead recommend using classes to define it.
   */
  isMember?: boolean;
  /**  @deprecated automationId, can use 'data-test-automation-id' replace */
  automationId?: string;
} & RcMenuItemClassesType &
  RcBaseProps<
    MuiMenuItemProps,
    'disableGutters' | 'button' | 'classes' | 'title' | 'dense' | 'divider'
  >;

const _RcMenuItem = forwardRef<any, RcMenuItemProps>((props, ref) => {
  const {
    icon,
    subtext,
    type,
    checked,
    value,
    symbol,
    children,
    disabled,
    classes,
    isMember,
    avatar,
    automationId,
    maxWidth,
    size,
    label,
    className,
    onMouseEnter,
    onClick,
    ...rest
  } = props;

  const isCheckedType = type === 'checked';

  const iconElement = useMemo(() => {
    if (React.isValidElement(avatar)) {
      return avatar;
    }
    if (symbol) {
      return <RcIcon size="small" symbol={symbol} />;
    }
    if (React.isValidElement(icon)) {
      return icon;
    }
    if (typeof icon === 'string') {
      return <RcIcon size="small">{icon}</RcIcon>;
    }
    return null;
  }, [avatar, icon, symbol]);

  const renderSubText = useMemo(() => {
    if (!isCheckedType) {
      return subtext;
    }
    return checked ? (
      <StyledRcIcon
        size="large"
        symbol={checkSvg}
        color={['interactive', 'f01']}
      />
    ) : (
      subtext
    );
  }, [checked, isCheckedType, subtext]);

  const [menuItemUUID, setMenuItemUUID] = useState<string | undefined>(
    undefined,
  );

  const menuContext = useContext(RcMenuContext);
  const menuListContext = useContext(RcMenuListContext);
  const subMenuContext = useContext(RcSubMenuContext);

  useEffect(() => {
    if (!menuListContext.isMenuList) {
      menuListContext.setIsMenuList();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    setMenuItemUUID(uniqueId('rc-menu-item-'));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleMenuItemMouseEnter = useEventCallback(
    (e: React.MouseEvent<HTMLLIElement>) => {
      onMouseEnter && onMouseEnter(e);

      if (!menuItemUUID) {
        return;
      }
      if (menuListContext.isMenuList) {
        menuListContext.setFocusedMenuItemId(menuItemUUID);
      } else {
        menuContext.setFocusedMenuItemId(menuItemUUID);
      }
    },
  );

  const handleMenuItemClick = useEventCallback(
    (e: React.MouseEvent<HTMLLIElement>) => {
      onClick && onClick(e);
      if (menuContext.autoClose) {
        menuContext.closeMenu(e, 'tabKeyDown');
      }
      if (subMenuContext.autoClose && subMenuContext.closeSubMenu) {
        subMenuContext.closeSubMenu();
      }
    },
  );

  const groupClasses = useMemo(
    () => combineClasses(RcMenuItemClasses, classes),
    [classes],
  );

  const toClasses = useMemo(
    () => omit(groupClasses!, ['checked', 'unchecked']),
    [groupClasses],
  );

  return (
    <StyledMenuItem
      data-test-automation-id={automationId}
      disabled={disabled}
      isMember={isMember}
      data-disabled={disabled}
      maxWidth={maxWidth}
      ref={ref}
      size={size}
      value={value}
      {...rest}
      classes={toClasses}
      onMouseEnter={handleMenuItemMouseEnter}
      onClick={handleMenuItemClick}
      className={clsx(className, {
        [groupClasses!.checked]: isCheckedType && checked,
        [groupClasses!.unchecked]: isCheckedType && !checked,
      })}
    >
      {iconElement && (
        <StyledMuiListItemIcon>{iconElement}</StyledMuiListItemIcon>
      )}
      {label && <StyledRcListItemLabel>{label}&nbsp;</StyledRcListItemLabel>}
      {children}
      {renderSubText}
    </StyledMenuItem>
  );
});

/** @release */
const RcMenuItem = styled(
  withDeprecatedCheck(
    withTooltip(_RcMenuItem),
    [
      {
        prop: 'isMember',
        time: '2021-1',
        comment:
          'isMember used for when hover on MenuItem, is cursor in shape of pointer. Please instead recommend using classes to define it.',
      },
      {
        prop: 'maxWidth',
        time: '2021-1',
        comment: 'Please instead recommend using classes to define it.',
      },
      {
        prop: 'label',
        time: '2021-1',
        comment: 'Please add label component and add to children prop',
      },
    ],
    'RcMenuItem',
  ),
)``;

RcMenuItem.displayName = 'RcMenuItem';

RcMenuItem.defaultProps = {
  size: 'medium',
};

export { RcMenuItem, RcMenuItemProps, RcMenuItemType, RcMenuItemClassesType };
