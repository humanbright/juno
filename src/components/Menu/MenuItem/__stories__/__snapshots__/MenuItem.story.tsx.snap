// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Menu/MenuItem MenuItem 1`] = `
Array [
  .c1 {
  padding: 0;
  margin: 0 8px 0 0;
}

.c1.RcListItemText-multiline {
  margin: 4px 8px 4px 0px;
}

.c1.RcListItemText-inset {
  padding-left: 56px;
}

.c1 .RcListItemText-primary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
}

.c1 .RcListItemText-secondary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
}

.c0 {
  outline: none;
  box-sizing: border-box;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c0 .RcMenuItem-gutters {
  padding-left: 16px;
  padding-right: 16px;
}

.c0.RcMenuItem-dense {
  padding-top: 4px;
  padding-bottom: 4px;
  min-height: auto;
  line-height: 22px;
  font-weight: 700;
}

<li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiMenuItem-gutters RcMenuItem-gutters MuiListItem-gutters MuiListItem-button"
    data-menuitem-id="menu-item-0"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    <div
      className="MuiListItemText-root c1"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary RcListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        MenuItem content
      </span>
    </div>
  </li>,
  <br />,
  .c1 {
  padding: 0;
  margin: 0 8px 0 0;
}

.c1.RcListItemText-multiline {
  margin: 4px 8px 4px 0px;
}

.c1.RcListItemText-inset {
  padding-left: 56px;
}

.c1 .RcListItemText-primary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
}

.c1 .RcListItemText-secondary {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-text-decoration-line: inherit;
  text-decoration-line: inherit;
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
}

.c0 {
  outline: none;
  box-sizing: border-box;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  height: auto;
  min-height: 32px;
  min-width: 112px;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c0 .RcMenuItem-gutters {
  padding-left: 16px;
  padding-right: 16px;
}

.c0.RcMenuItem-dense {
  padding-top: 4px;
  padding-bottom: 4px;
  min-height: auto;
  line-height: 22px;
  font-weight: 700;
}

<li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root c0 MuiMenuItem-gutters RcMenuItem-gutters MuiListItem-gutters MuiListItem-button"
    data-menuitem-id="menu-item-0"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    <div
      className="MuiListItemText-root c1 MuiListItemText-multiline RcListItemText-multiline"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary RcListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        MenuItem content
      </span>
      <p
        className="MuiTypography-root MuiListItemText-secondary RcListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Secondary content
      </p>
    </div>
  </li>,
]
`;
